app/scripts/views/mixins/avatar-mixin.js

```diff
@@ -62,11 +62,10 @@ define([
 
     // Makes sure the account with uid has an uptodate image cache.
     // This should be called after fetching the current profile image.
-    _updateCachedProfileImage: function (uid, profileImage) {
-      var cachedAccount = this.user.getAccountByUid(uid);
-      if (! cachedAccount.isDefault()) {
-        cachedAccount.setProfileImage(profileImage);
-        this.user.setAccount(cachedAccount);
+    _updateCachedProfileImage: function (account, profileImage) {
+      if (! account.isDefault()) {
+        account.setProfileImage(profileImage);
+        this.user.setAccount(account);
```

Comment:
Since an `account` is a model, could a `user` listen for a [change event](http://backbonejs.org/#Model-set) on the account, and then call `self._persistAccount` in the handler?

```js
var self = this;
account.on('change', function () {
  self._persistAccount(account.toPersistentJSON());
});
```
or 
```js
var self = this;
account.on('change', function () {
  self.setAccount(account);
});
```

That way the views wouldn't have to worry about setting the account on the user every time they make an update.